### PR TITLE
[Feature][MaterializedView] add AlterMaterializedViewStmt for MaterializedView

### LINKED ISSUE #4782: [MaterializedView] add AlterMaterializedViewStmt for MaterializedView [P2]
## Feature request

**Is your feature request related to a problem? Please describe.**

to #3680

**Describe the solution you'd like**
 
add AlterMaterializedViewStmt for new MaterializedView
1. modify time interval

### PR DESCRIPTION
**Pull request by zhuxt2015**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #4782

### CODE REVIEW

**Comment by zhuxt2015** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 142 / 213 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|4|25.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|6|33.33%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|4|11|36.36%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/alter/Alter.java|44|79|55.70%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|9|11|81.82%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|16|81.25%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|27|32|84.38%|
|:large_blue_circle:|com/starrocks/persist/ChangeMaterializedViewRefreshSchemeLog.java|13|15|86.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AlterMaterializedViewStatement.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/persist/RenameMaterializedViewLog.java|13|13|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/IntervalLiteral.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -53,6 +60,15 @@ public static void analyze(StatementBase stmt, ConnectContext session) {
 
     static class MaterializedViewAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
 
+        public enum RefreshTimeUnit {
+            YEAR,
+            MONTH,
+            WEEK,
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:66]: It is better that this version does not support WEEK first

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java [hunk 2]
```diff
@@ -26,18 +29,21 @@
...
     public static class AsyncRefreshContext {
         // base table id -> (partitionid -> visible version)
         @SerializedName(value = "baseTableVisibleVersionMap")
         public Map<Long, Map<Long, Long>> baseTableVisibleVersionMap;
 
+        @SerializedName(value = "starTime")
+        private long startTime;
+
+        @SerializedName(value = "intervalLiteral")
+        private IntervalLiteral intervalLiteral;
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java:41]: better not use IntervalLiteral to serialize and deserialize, use basic type

**Comment by zhuxt2015** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 147 / 216 (68.06%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|6|33.33%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|4|11|36.36%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/alter/Alter.java|47|81|58.02%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|16|81.25%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|26|31|83.87%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|13|15|86.67%|
|:large_blue_circle:|com/starrocks/persist/ChangeMaterializedViewRefreshSchemeLog.java|13|15|86.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AlterMaterializedViewStatement.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/persist/RenameMaterializedViewLog.java|13|13|100.00%|

**Review by Astralidea — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**